### PR TITLE
libgit2 test: use a valid user/pass parameter to test EAUTH instead of OS:ERROR

### PR DESCRIPTION
**Pull request by vtjnash**
this should fix the appveyor failures resulting from upgrading libgit2

### CODE REVIEW

**Comment by tkelman** [comment]
Doesn't make a lot of sense that the old inputs would cause a different error on different platforms and versions of libgit2. @omus opinion on why that might be?

**Comment by omus** [comment]
I agree that this doesn't make much sense. Maybe something was added into v0.26.0 that was Windows specific which is looking checking for blank usernames and/or passwords? Did Appveyor start using LibGit2 v0.26.0 only when #22614 was merged?

**Comment by tkelman** [comment]
only after, had to make it into nightlies since appveyor doesn't build most deps from source

**Comment by tkelman** [comment]
Bisected within libgit2, change in behavior was introduced by https://github.com/libgit2/libgit2/pull/4055. Not entirely sure if this side-effect was intentional there, may be worth reporting the newly introduced platform discrepancy upstream.

**Comment by tkelman** [comment]
Could we leave the existing test (in addition to something new like this) but with a test_broken for is_windows() and libgit2 version >= 0.26 until we're sure what the intended behavior from libgit2 should be?

File: test/libgit2-online.jl [hunk 1]
```diff
@@ -22,13 +22,32 @@ mktempdir() do dir
             try
                 repo_path = joinpath(dir, "Example2")
                 # credentials are required because github tries to authenticate on unknown repo
-                cred = LibGit2.UserPasswordCredentials("","") # empty credentials cause authentication error
+                cred = LibGit2.UserPasswordCredentials("JeffBezanson", "hunter2") # make sure Jeff is using a good password :)
```
Comment (tkelman) [inline test/libgit2-online.jl]: cc @JeffBezanson in case you'd rather have this do something different

**Comment by omus** [comment]
I'd like to get this resolved soon. If we want to spend more time tweaking this PR we should just merge the conditional changes so the CI is functional again.

File: test/libgit2-online.jl (hunk 1 above)
Comment (JeffBezanson) [inline test/libgit2-online.jl]: Hey, how did you get my password?!! :)
Doesn't matter to me, I just want AV to work again.

**Comment by yuyichao** [comment]
Given https://github.com/JuliaLang/julia/pull/22681#discussion_r126287828 I'll just merge this. Anyone that has strong feeling about doing this different should have commented here or open a new pr that tweak this later.

**Pull request merged**